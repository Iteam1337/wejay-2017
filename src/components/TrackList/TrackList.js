// @flow

import * as WejayApi from '__generated__/types.flow'
import React from 'react'
import Track from '../Track/Track'
import styled from 'styled-components'

type TrackListProps = {
  tracks: WejayApi.TrackInfoFragment[],
}

const TrackListWrap = styled.section``

const TrackList = ({ tracks }: TrackListProps) => {
  if (!tracks || tracks.length === 0) {
    return <div className="EmptyState">Queue is empty</div>
  }

  return (
    <TrackListWrap>
      {tracks.map(track => <Track key={track.spotifyUri} track={track} />)}
    </TrackListWrap>
  )
}

export default TrackList
