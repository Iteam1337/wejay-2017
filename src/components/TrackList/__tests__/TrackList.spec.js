import React from 'react'
import { shallow } from 'enzyme'
import TrackList from '../TrackList'

describe('components/TrackList', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <TrackList
        tracks={[
          {
            name: 'Circles',
            spotifyUri: 1337,
          },
        ]}
      />
    )
  })

  it('can render', () => {
    expect(component).toMatchSnapshot()
  })

  it('should display empty state when no TrackList', () => {
    component.setProps({
      tracks: [],
    })

    expect(component.find('.EmptyState')).toMatchSnapshot()
  })
})
