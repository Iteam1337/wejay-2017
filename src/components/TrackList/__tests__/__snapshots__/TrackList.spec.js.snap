// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TrackList can render 1`] = `
<styled.section>
  <TrackItem
    key="1337"
    track={
      Object {
        "name": "Circles",
        "spotifyUri": 1337,
      }
    }
  />
</styled.section>
`;

exports[`components/TrackList should display empty state when no TrackList 1`] = `
<div
  className="EmptyState"
>
  Queue is empty
</div>
`;
